# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.h
# Commit: b3e786ad051c
# Full Hash: b3e786ad051cb83344b111e5cb8a19303dd6a77e
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1789849
# File Overlap Count: 2
# Description:
#   Bug 1789849 - Use a separate DrawTargetSkia for rendering clip masks. r=aosmond
#   
#   Previously we were reusing the framebuffer's Skia DT to render the clip mask.
#   This was the path of least resistance since SkCanvas does not allow exporting
#   clip information, and there is no way to reset the bitmap storage inside an
# ==============================================================================

diff -r 0b582207c30d -r b3e786ad051c dom/canvas/DrawTargetWebgl.h
--- a/dom/canvas/DrawTargetWebgl.h	Mon Sep 12 15:26:02 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.h	Mon Sep 12 15:26:03 2022 +0000
@@ -8,6 +8,7 @@
 #define _MOZILLA_GFX_DRAWTARGETWEBGL_H
 
 #include "mozilla/gfx/2D.h"
+#include "mozilla/gfx/PathSkia.h"
 #include "mozilla/LinkedList.h"
 #include "mozilla/WeakPtr.h"
 #include "mozilla/ThreadLocal.h"
@@ -94,6 +95,16 @@
 
   RefPtr<TextureHandle> mSnapshotTexture;
 
+  // Store a log of clips currently pushed so that they can be used to init
+  // the clip state of temporary DTs.
+  struct ClipStack {
+    Matrix mTransform;
+    Rect mRect;
+    RefPtr<const Path> mPath;
+  };
+
+  std::vector<ClipStack> mClipStack;
+
   // UsageProfile stores per-frame counters for significant profiling events
   // that assist in determining whether acceleration should still be used for
   // a Canvas2D user.
@@ -174,7 +185,7 @@
     RefPtr<WebGLFramebufferJS> mScratchFramebuffer;
     // Buffer filled with zero data for initializing textures.
     RefPtr<WebGLBufferJS> mZeroBuffer;
-    IntSize mZeroSize;
+    size_t mZeroSize = 0;
     // 1x1 texture with solid white mask for disabling clipping
     RefPtr<WebGLTextureJS> mNoClipMask;
 
